libs/execution_context/ebpf_maps.c

```diff
@@ -1209,6 +1340,74 @@ _delete_lru_hash_map(_In_ _Post_invalid_ ebpf_core_map_t* map)
 static ebpf_result_t
 _delete_hash_map_entry(_Inout_ ebpf_core_map_t* map, _In_ const uint8_t* key);
 
+/**
+ * @brief Walk the cold lists, removing secondary keys and finding the oldest primary key.
+ *
+ * @param[in] lru_map
+ * @return Either the oldest primary key or NULL if no primary keys are present.
+ */
+static ebpf_lru_entry_t*
+_reap_lru_cold_lists(ebpf_core_lru_map_t* lru_map)
+{
+    ebpf_lru_entry_t* oldest_entry = NULL;
+    uint64_t oldest_timestamp = UINT64_MAX;
+    bool non_empty_partition_found = true;
+
+    // Scan all partitions until we find a non-empty cold list or we have an oldest entry.
+    while (!oldest_entry && non_empty_partition_found) {
```

Comment:
Isn't the whole block a critical section to lock, starting from scan and update?

Reply:
No, CPUs are free to execute this concurrently. This may result in multiple threads attempting to evict the same key, but this doesn't really justify the cost of adding a map wide lock here.